File: src/test/java/ting/leetcode/java/hashtable/JewelsAndStonesTest.java
package ting.leetcode.java.hashtable;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class JewelsAndStonesTest {
    private JewelsAndStones jewelsAndStones;

    @BeforeEach
    void setUp() {
        jewelsAndStones = new JewelsAndStones();
    }

    @Test
    void numJewelsInStones() {
        assertEquals(jewelsAndStones.numJewelsInStones("aA", "aAAbbbb"), 3);
        assertEquals(jewelsAndStones.numJewelsInStones("z", "ZZ"), 0);
    }

    @Test
    void numJewelsInStonesWithSet() {
        assertEquals(jewelsAndStones.numJewelsInStonesWithList("aA", "aAAbbbb"), 3);
        assertEquals(jewelsAndStones.numJewelsInStonesWithList("z", "ZZ"), 0);
    }
}

File: src/test/kotlin/ting/leetcode/hashtable/JewelsAndStonesTest.kt
package ting.leetcode.hashtable

import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

internal class JewelsAndStonesTest {
    private lateinit var jewelsAndStones: JewelsAndStones

    @BeforeEach
    fun setUp() {
        jewelsAndStones = JewelsAndStones()
    }

    @Test
    fun numJewelsInStones() {
        assertEquals(jewelsAndStones.numJewelsInStones("aA", "aAAbbbb"), 3)
        assertEquals(jewelsAndStones.numJewelsInStones("z", "ZZ"), 0)
    }

    @Test
    fun numJewelsInStonesWithMap() {
        assertEquals(jewelsAndStones.numJewelsInStonesWithMap("aA", "aAAbbbb"), 3)
        assertEquals(jewelsAndStones.numJewelsInStonesWithMap("z", "ZZ"), 0)
    }
}